addons/delivery/tests/test_delivery_cost.py

```diff
@@ -327,3 +332,35 @@ def test_add_carrier_on_picking(self):
             {'product_id': self.product_2.id, 'is_delivery': False, 'product_uom_qty': 1, 'qty_delivered': 1},
             {'product_id': self.normal_delivery.product_id.id, 'is_delivery': True, 'product_uom_qty': 1, 'qty_delivered': 0},
         ])
+
+    def test_delivery_carrier_weight(self):
+        '''
+        Test If in SO order line has same quantity in 2 line but one of them is nagetive,
+        so it'll count that nagetive quantity in shipping weight or not.
+        '''
+        sale_order = self.SaleOrder.create({
+            'partner_id': self.partner_18.id,
+            'name': 'Cable Management Box',
+            'order_line': [(0, 0, {
+                    'product_id': self.product_cable_management_box.id,
+                    'product_uom_qty': 2,
+                    'product_uom': self.product_uom_unit.id,
+                    'price_unit': 750.00,
+                }),
+                (0, 0, {
+                    'name': 'Cable Management Box',
+                    'product_id': self.product_cable_management_box.id,
+                    'product_uom_qty': -2,
+                    'product_uom': self.product_uom_unit.id,
+                    'price_unit': 750.00,
+                })],
+        })
+
+        delivery_wizard = Form(self.env['choose.delivery.carrier'].with_context({
+            'default_order_id': sale_order.id,
+            'default_carrier_id': self.env.ref('delivery_fedex.delivery_carrier_fedex_us').id,
+        }))
+
+        delivery_wizard.save()
+        shipping_weight = sale_order._get_estimated_weight()
+        self.assertEqual(shipping_weight, 2.0, "Actual weight and shpping weight is different")
```

Comment:
```suggestion
    def test_estimated_weight(self):
        """
        Test that negative qty SO lines are not included in the estimated weight calculation
        of delivery carriers (since it's used when calculating their rates).
        """
        sale_order = self.SaleOrder.create({
            'partner_id': self.partner_18.id,
            'name': 'SO - neg qty',
            'order_line': [
                (0, 0, {
                    'product_id': self.product_4.id,
                    'product_uom_qty': 1,
                    'product_uom': self.product_uom_unit.id,
                }),
                (0, 0, {
                    'product_id': self.product_2.id,
                    'product_uom_qty': -1,
                    'product_uom': self.product_uom_unit.id,
                })],
        })

        shipping_weight = sale_order._get_estimated_weight()
        self.assertEqual(shipping_weight, self.product_4.weight, "Only positive quantity products' weights should be included in estimated weight")
```